src/throttler/throttler.ts

```diff
@@ -140,17 +168,47 @@ export abstract class Throttler<T, R> {
     this.handle(this.nextWaitingTaskIndex());
   }
 
-  async handle(cursorIndex: number): Promise<void> {
+  stats(): ThrottlerStats {
+    return {
+      max: this.max,
+      min: this.min,
+      avg: this.avg,
+      active: this.active,
+      complete: this.complete,
+      success: this.success,
+      errored: this.errored,
+      retried: this.retried,
+      total: this.total,
+      elapsed: Date.now() - this.startTime,
+    };
+  }
+
+  taskName(cursorIndex: number): string {
     const taskData = this.taskDataMap.get(cursorIndex);
     if (!taskData) {
-      throw new Error(`taskData.get(${cursorIndex}) does not exist`);
+      return "finished task";
```

Comment:
We can't return "finished task" here, since that won't always be true (eg. when requesting a cursorIndex that never existed). Maybe keep the thrown error that existed before, but update the message to say "{} does not exist. It may be a task that has already completed.".

Reply:
Aborted since it's not a change in this PR